### PR TITLE
Trying to refactor cinemaHall index  to moneteway

### PR DESCRIPTION
**Pull request by agusiol**
I am trying to refactor CinemaHall index  to moneteway, but can't solve the following error:
Action Controller: Exception caught 2021-05-25 14-38-27

### CODE REVIEW

**Comment by wmaciejak** [comment]
Could you try to use:
```
@cinema_halls = ::UseCases::CinemaHalls::FetchForCinemaHalls.new.call
```

Please tell if it solved something. This part `::` means that rails should try to find UseCase class out of controller scope. I'm not sure why you have this problem here, but it might be a solution

**Comment by jandudulski** [comment]
The problem is because the structure doesn't follow [zeitwerk](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html) rules. You have at least two simple options to make it work out of the box:

1. remove `UseCases` module
2. add another `use_cases` directory in the `use_cases` directory

TL;DR - first depth of directories under `/app` should not generate any namespace - they're just for organization. E.g.:

`/app/foo/bar.rb` should define `class Bar`
`/app/foo/foo/bar.rb` should define `class Foo::Bar`

**Comment by wmaciejak** [comment]
> The problem is because the structure doesn't follow [zeitwerk](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html) rules. You have at least two simple options to make it work out of the box:
> 
> 1. remove `UseCases` module
> 2. add another `use_cases` directory in the `use_cases` directory
> 
> TL;DR - first depth of directories under `/app` should not generate any namespace - they're just for organization. E.g.:
> 
> `/app/foo/bar.rb` should define `class Bar`
> `/app/foo/foo/bar.rb` should define `class Foo::Bar`

Today I Learned. I didn't know that zeitwerk works in that way. Thank you @jandudulski

File: app/repositories/cinema_hall_repository.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            @adapter = adapter
+        end
+
+        def fetch_all_with_columns(columns:)
+            adapter.select(*columns)
+        end
+
+        # def find_all
+        #     adapter.all
+        end
```
Comment (w0st) [inline app/repositories/cinema_hall_repository.rb:15]: I am afraid it's a syntax error (incorrect code, I see `end` but I don't see a 'def` - it's commented.

Generally, it's really bad to keep commented code. It's not the purpose of comments.

https://kentcdodds.com/blog/please-dont-commit-commented-out-code

File: app/controllers/cinema_halls_controller.rb [hunk 2]
```diff
@@ -1,55 +1,60 @@
...
-    end
-
-    def show
-        @cinema_hall = {
-            id: @cinema_hall.id
-            name: @cinema_hall.name,
-            cpacity: @cinema_hall.capacity
-        }
-        render json: @cinema_hall
+        @cinema_halls = UseCases::CinemaHalls::FetchForCinemaHalls.new.call
```
Comment (w0st) [inline app/controllers/cinema_halls_controller.rb:3]: Compare the indentation here with any other file in the repository committed before, for example: https://github.com/agusiol/montecinema/blob/main/config/application.rb#L23 . As you can spot your file indentation has 4 spaces, the other one has an indentation of 2 spaces. You should keep it consistent for many reasons, mostly to avoid "noise" in the repository. 

Here you can find how to set it in VS Code: https://stackoverflow.com/questions/34174207/how-to-change-indentation-in-visual-studio-code

**Review by w0st — COMMENTED** [review]
The `.DS_Store` as you can check contains your personal setting on your machine. It shouldn't be a part of the repository. It could lead to false conflict (someone else could have different preferences on their machine). Files like that should be a part of .gitignore as was discussed during the Github intro. I propose to add such files to your global .gitignore. Here you can find more about it: https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/ignoring-files

**Comment by w0st** [comment]
> The problem is because the structure doesn't follow [zeitwerk](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html) rules. You have at least two simple options to make it work out of the box:
> 
> 1. remove `UseCases` module
> 2. add another `use_cases` directory in the `use_cases` directory
> 
> TL;DR - first depth of directories under `/app` should not generate any namespace - they're just for organization. E.g.:
> 
> `/app/foo/bar.rb` should define `class Bar`
> `/app/foo/foo/bar.rb` should define `class Foo::Bar`

Another option should be adding to `config/application.rb` something like that if you really know what you do:
```
config.config.autoload_paths << Rails.root.join('app/use_cases')
```

File: .gitignore [hunk 3]
```diff
@@ -25,3 +25,6 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+
+#Ignor DS_Store
+**/.DS_Store
```
Comment (jandudulski) [inline .gitignore:30]: It's good idea to setup [global git ignore](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) for such files.
